core/include/detray/definitions/grid_axis.hpp

```diff
@@ -44,4 +47,51 @@ enum class binning {
     e_irregular = 1,
 };
 
+DETRAY_HOST inline std::ostream& operator<<(std::ostream& os, bounds b) {
+    switch (b) {
+        using enum bounds;
+        case e_open:
+            os << "e_open";
+            break;
+        case e_closed:
+            os << "e_closed";
+            break;
+        case e_circular:
+            os << "e_circular";
+            break;
+    }
+    return os;
+}
+
+DETRAY_HOST inline std::ostream& operator<<(std::ostream& os, label l) {
+    switch (l) {
+        using enum label;
+        case e_x:
+            // e_r and e_rphi have same value (0)
+            os << "e_x/e_r/e_rphi";
+            break;
+        case e_y:
+            // e_phi and e_cyl_z have same value (1)
+            os << "e_y/e_phi/e_cyl_z";
+            break;
+        case e_z:
+            os << "e_z";
+            break;
+    }
+    return os;
+}
+
+DETRAY_HOST inline std::ostream& operator<<(std::ostream& os, binning b) {
+    switch (b) {
+        using enum binning;
+        case e_regular:
+            os << "e_regular";
+            break;
+        case e_irregular:
+            os << "e_irregular";
+            break;
+    }
+    return os;
+}
+
```

Comment:
This means we can remove the `static_cast` here, as well, right?
https://github.com/acts-project/detray/blob/5a0ad05d53b49122be226adc91217e1934e8ca00/core/include/detray/utils/grid/detail/axis.hpp#L188-L190